### DISCUSSION TITLE
Moving the msys64 folder

### DISCUSSION THREAD

**Discussion by goyalyashpal** [msys2/MSYS2-packages#3504] [Q&A] [answered]
* msys was installed in c drive
* so, the msys64 folder had acquired _HUGE_ size
* the C drive is on a separate disk, which is running overflow now

So:
* i'd like to move the `msys64` folder to a new location on a separate drive
* and also change the default location used by programs inside it to that location
* and would like to know if there can be other implications of it

**Comment by Biswa96** [comment, marked as answer]
Instead of moving the whole installation, I would suggest to install msys2 again in the separate partition/drive. Then install all the packages in new installation by copying the packages list installed in previous one. And move any config file as usual. At last delete all the package in old one and remove the old install.

**Reply by goyalyashpal** [reply]
> Then install all the packages in new installation by copying the packages list installed in previous one

how?

**Reply by Biswa96** [reply]
`pacman -Qqs` can show the list of installed packages.

**Reply by goyalyashpal** [reply]
i mean yes, this can list those, but how to "install them by copying that package list" ?

**Reply by Biswa96** [reply]
It's simple shell tricks.

1. In old installation, save the package list in a text file: `pacman -Qqs > test.txt`
2. In new installation, install the packages from that list: `cat test.txt | pacman -S -`

**Reply by goyalyashpal** [reply]
To self: will tackle this next after gathering the files

Things to gather:
- [config files] `.bashrc` etc (multiple locations, a bit tough search)
- history: 2 locations, easy
- pacman.log : 1 location , super easy (`msys??/var/log` i guess)

[config files]: https://github.com/msys2/MSYS2-packages/issues/1798#issuecomment-1368395574

are there any other kind of log files? which might be useful

**Reply by goyalyashpal** [reply]
so, the concerned config files are only in home/<user> directory, the `etc` ones were untouched. that's better.
history files were also only on home dir itslelf

----
I will be installing packages as i go :) currently, 
- 'can only think of: `gcc make` `clang`. 
- will seee for `pactoys` and
- `python wslbridge2` someday later.

also stumbled upon an article on "[Re-installing MSYS2]" ([semi-permalink])
- 5th pt is important: to save server bandwidth, reuse old cached package directory
`msys??.old\var\cache\pacman\pkg` (370 MiB for me)
- other than that, i think it's old (see command for updating core package lol)
`pacman --needed -S bash pacman pacman-mirrors msys2-runtime`
- **'23.03.11:** this removal change [dbb67d2] for same command `-S --needed` at [index.md#L60] (via blame)
is very very welcome - it keeps the footprint light for both the installation size & `-Syu` upgrades bandwidth requirement

[Re-installing MSYS2]: https://www.msys2.org/wiki/MSYS2-reinstallation/
[semi-permalink]: https://github.com/msys2/msys2.github.io/blob/48e6d751dc784a7f02a597c7ce98e00ffb297f48/web/wiki/MSYS2-reinstallation.md


----
2023.03.11
- some more nice tools i found: <tt>[lftp] unzip [colormake-git]</tt> 
- i also noticed that <tt>man</tt> is missing now, it seems to be `man-db` via `pacman -F man{,.exe}`
<sub>thanks to that discussion https://github.com/msys2/MSYS2-packages/discussions/3593#discussioncomment-5187899</sub>
- some others: `astyle:x qrencode:x` (using pacboy translation codes)
- some others: (not installed yet due to confusions:) <tt>[ctags] [cppcheck]</tt> (kate-editor), and <tt>[clang-tools-extra]</tt> (for `clangd.exe` LS)

[dbb67d2]: https://github.com/msys2/msys2.github.io/commit/dbb67d25c9164a86caf993f76821070b61259438
[index.md#L60]: https://github.com/msys2/msys2.github.io/blame/source/web/index.md#L60

[ctags]: https://packages.msys2.org/search?t=pkg&q=ctags
[cppcheck]: https://packages.msys2.org/search?t=pkg&q=cppcheck
[clang-tools-extra]: https://packages.msys2.org/search?t=binpkg&q=clang+tools+extra

[lftp]: https://github.com/msys2/MSYS2-packages/discussions/3534#discussioncomment-5223971
[colormake-git]: https://github.com/renatosilva/colormake
